components/script/dom/blob.rs

```diff
@@ -173,7 +165,31 @@ impl Blob {
     /// Get a FileID representing the Blob content,
     /// used by URL.createObjectURL
     pub fn get_blob_url_id(&self) -> SelectedFileId {
-        match *self.blob_impl.borrow() {
+        let opt_sliced_parent = match *self.blob_impl.borrow() {
+            BlobImpl::Sliced(ref parent, ref rel_pos) => {
+                Some((parent.promote(/* set_valid is */ false), rel_pos.clone(), parent.Size()))
+            }
+            _ => None
+        };
+
+        match opt_sliced_parent {
+            Some((parent_id, rel_pos, size)) => self.create_sliced_url_id(&parent_id, &rel_pos, size),
```

Comment:
This step is basically casting the script-side slicing relation to file manager side.